pydantic/dataclasses.py

```diff
@@ -22,6 +22,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         def __validate__(cls, v: Any) -> 'DataclassType':
             pass
 
+        def __call__(self, *args: Any, **kwargs: Any) -> 'DataclassType':
```

Comment:
Should this use a TypeVar?

Reply:
Did you expect this?
```python
    def __call__(self: T, *args: Any, **kwargs: Any) -> T:
        pass
```
Also, Did you want to replace other `-> 'DataclassType'`  like `__init__` ?

Reply:
Yeah, that's what I meant, I think it would be better (assuming that is the right signature, which I think it is?).

I don't think the `__init__` signature needs to change, but maybe the `_validate_dataclass` one should.

(Any other places where `'DataclassType'` is currently used probably should; I believe this is what is done for `BaseModel` -- I don't see why it shouldn't be that way for the dataclasses.)

Also, I would recommend using something like `DataclassT` as the name of the `TypeVar`, to make it slightly more obvious what's happening, and give it `bound=DataclassType`. (The TypeVar for `BaseModel` in `pydantic.main` is called `Model`; I think that's nice.)